src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceProgress.cs

```diff
@@ -28,10 +29,22 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
             // destroy the data structures representing outstanding progress records
             // take down and destroy the progress display
 
+            if (_progPaneUpdateTimer != null)
+            {
+                // Stop update a progress pane and destroy timer
+                _progPaneUpdateTimer.Dispose();
+                _progPaneUpdateTimer = null;
+            }
+            // We don't reset 'progPaneUpdateFlag = false' here
+            // because `HandleIncomingProgressRecord` will be init 'progPaneUpdateFlag' in any way.
+            // (Also the timer callback can still set it to 'true' accidentally)
```

Comment:
This comment is not clear enough. How about
```
// We don't set 'progPaneUpdateFlag = 0' here, because:
// 1. According to MSDN, the timer callback can occur after the Dispose() method has been called. So we cannot guarantee the flag is truly set to 0.
// 2. When creating a new timer in 'HandleIncomingProgressRecord', we will set the flag to 1 anyway
```